src/sema/expression.rs

```diff
@@ -3905,11 +3937,19 @@ fn equal(
 
     let ty = coerce(&left_type, &l.loc(), &right_type, &r.loc(), ns, diagnostics)?;
 
-    Ok(Expression::Equal(
+    let expr = Expression::Equal(
         *loc,
         Box::new(left.cast(&l.loc(), &ty, true, ns, diagnostics)?),
         Box::new(right.cast(&r.loc(), &ty, true, ns, diagnostics)?),
-    ))
+    );
+
+    if ty.is_rational() {
+        if let Err(diag) = eval_const_rational(&expr, ns) {
+            diagnostics.push(diag);
+        }
+    }
+
+    Ok(expr)
```

Comment:
What about `NotEqual`?

Reply:
`NotEqual` is implement with `fn equal()` , so this works fine too. I've added a test for this.